modules/administration-guide/partials/proc_adding-or-removing-vscode-extensions-in-the-embedded-open-vsx-registry-instance.adoc

```diff
@@ -0,0 +1,83 @@
+
+You can add or remove your choice of pass:[<!-- vale RedHat.Spelling = NO -->] pass:[<!-- vale RedHat.Spelling = NO -->] VS Code pass:[<!-- vale RedHat.Spelling = YES -->] pass:[<!-- vale RedHat.Spelling = YES -->] extensions in the embedded Open VSX registry instance that is cached by {prod-short} to support air-gapped, offline, and proxy-restricted environments.
+
+This will create a custom build of the Open VSX registry, which can be used in your organization’s workspaces. 
+
+NOTE: A subscription update reverts your `CheCluster` to default settings, and you must reapply your changes to `CheCluster`. 
+
+.Procedure
+
+include::example$snip_{project-context}-steps-to-add-or-remove-extensions-in-openvsx-sync-json.adoc[]
+
+** If the extension is published on link:https://open-vsx.org[open-vsx.org], you can add it by including an object with the extension `id` in the format __<published_by>__.__<unique_identifier>__ that you can collect from the extension page on link:https://open-vsx.org[open-vsx.org]:
++
+[source,json,subs="+quotes"]
+----
+    {
+        "id": "__<published_by>__.__<unique_identifier>__"
+    }
+----
++
+[TIP]
+====
+The latest extension version on link:https://open-vsx.org[open-vsx.org] is the default. Alternatively, you can add `"version": "__<extension_version>__"` on a new line to specify a version.
+====
+
+** If the extension is only available from link:https://marketplace.visualstudio.com/VSCode[Microsoft Visual Studio Marketplace], but not link:https://open-vsx.org[Open VSX], you can ask the extension publisher to also publish it on link:https://open-vsx.org[open-vsx.org] according to the link:https://github.com/eclipse/openvsx/wiki/Publishing-Extensions#how-to-publish-an-extension[instructions] and the link:https://github.com/marketplace/actions/publish-vs-code-extension[GitHub action].
++
+[TIP]
+====
+* If the extension publisher is not available for publishing the extension on link:https://open-vsx.org[open-vsx.org], and if there is no Open VSX equivalent of the extension, consider link:https://github.com/open-vsx/publish-extensions/issues[notifying] the Open VSX team.
+
+* If you have a closed-source extension or an extension developed only for internal use in your company, you can add the extension directly from a `.vsix` file that is accessible from a URL while building the plugin registry container:
++
+[source,json,subs="+quotes"]
+----
+    {
+        "id": "__<published_by>__.__<unique_identifier>__",
+        "download": "__<url_to_download_vsix_file>__",
+        "version": "__<extension_version>__"
+    }
+----
+====
++
+WARNING: The link:https://aka.ms/vsmarketplace-ToU[Microsoft Visual Studio Marketplace Terms of Use] do not permit downloading extensions from link:https://marketplace.visualstudio.com/VSCode[Microsoft Visual Studio Marketplace] for use in Microsoft Visual Studio Code - Open Source that is loaded in your organization's workspaces.
```

Comment:
```suggestion
WARNING: The link:https://aka.ms/vsmarketplace-ToU[Terms of Use] for the link:https://marketplace.visualstudio.com/VSCode[Microsoft Visual Studio Marketplace] prevent using extensions in {prod-short}.
```

Reply:
@nickboldt:

1. I phrased the first sentence the way I did for a reason: My sentence says that customers are not permitted to download extensions from [VS Marketplace]. Your suggested sentence reads that extensions from [VS Marketplace]  cannot be used [at all] because of the Terms of Use. But, correct me if I'm wrong, those same extensions can also be published by their respective authors or forkers elsewhere and downloaded from elsewhere. So I tried avoiding a phrasing similar to yours because those extensions, if concurrently published elsewhere, are effectively available to our customers but just cannot be downloaded from [VS Marketplace] in particular. Don't you think?

2. I find the second sentence in your suggestion confusing for a few reasons: (A) _Only Microsoft products are permitted_ is confusing because we say we offer Microsoft Visual Studio Code - Open Source. (B) You mention a URL, but it takes some thinking to figure out which URL you mean. (C) _Only Microsoft products are permitted to use ._ sounds to me like those products are the users who are permitted to do something (maybe it's because I'm not a native speaker, but at the same time, not every reader of the docs is).

And my last point: :D Since it was Microsoft that composed that restrictive license (i.e., Microsoft does not allow using those extensions in Microsoft Visual Studio Code - Open Source), is it now our job to sugarcoat it?

BTW, I copied the TOU title for the link as it is in Microsoft's original PDF doc. So however awkward it may read, it is from the original text.

Can you live with my text as it is, without changes?

![image](https://user-images.githubusercontent.com/78564304/209223575-8da59325-f9c2-4667-b58f-d8c6e0ec8c11.png)

Reply:
> Your suggested sentence reads that extensions from [VS Marketplace] cannot be used [at all] because of the Terms of Use.

Yes. That is correct.

If a maintainer chooses to publish the extension to another marketplace (eg., open-vsx) then that's a DIFFERENT URL governed by a DIFFERENT TOU. 

TL;DR: DO NOT DOWNLOAD from M$ or face the wrath of their lawyers. :D

Reply:
"Microsoft Visual Studio Code - Open Source" is not a MS product.

It's a pile of open source code, which WE build into a product. It is governed by a different license and different TOU.

Reply:
I still prefer my statement as it more accurately explains that we cannot willfully download from the MS marketplace:

The Terms of Use for the Microsoft Visual Studio Marketplace prevent using extensions from that URL in {prod-short}. Only Microsoft products are permitted to use files from the Microsoft Visual Studio Marketplace.

The TOU does NOT mention MS VS Code - Open Source in its list of limited allowed products, which to me implies that the Open Source repo != a product. 

> "Microsoft Visual Studio, Visual Studio for Mac, Visual Studio Code, GitHub Codespaces, Azure DevOps, Azure DevOps Server, and successor products and services (the “In-Scope Products and Services”) offered by us and GitHub, Inc. (“GitHub”).

Reply:
I would just suggest Che admins and users to be mindful of the Microsoft Marketplace ToU that limits access to its extensions. I would not spend time interpreting the ToU or even copying it's content in Che documentation as it can change.

Reply:
```
Please read the >The Terms of Use< for the >Microsoft Visual Studio Marketplace< before using its resources.
```

Something simple like that?

Reply:
Yes I like that.

Reply:
suggestion updated to a `NOTE` instead of a `WARNING`. @max-cx are you OK with this new simpler pointer to the ToU and passing the action to the end user, rather than interpreting the ToU here?